cpp/crawl_v1_executor.cpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <vector>
+
+#include <boost/regex.hpp>
+
+#include <curl/curl.h>
+
+#include <stout/os.hpp>
+
+#include "rendler_helper.hpp"
+#include "rendler_v1_executor.hpp"
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+using mesos::vectorToString;
+
+
+static int writer(char *data, size_t size, size_t nmemb, string *writerData)
+{
+  assert(writerData != NULL);
+  writerData->append(data, size*nmemb);
+  return size * nmemb;
+}
+
+
+class CrawlV1Executor : public RendlerV1Executor
+{
+public:
+  CrawlV1Executor(const FrameworkID& _frameworkId,
+                  const ExecutorID& _executorId)
+    : RendlerV1Executor("CrawlV1Executor", _frameworkId, _executorId) {}
+
+  virtual ~CrawlV1Executor() {}
+
+protected:
+  void runTask(const TaskInfo& task) override
+  {
+    string url = task.data();
+    cout << "Running crawl task " << task.task_id().value()
+         << " Fetch: " << url;
+
+    string buffer;
+    vector<string> result;
+    result.push_back(task.task_id().value());
+    result.push_back(url);
+
+    CURL *conn;
```

Comment:
I'm assuming you're talking about running curl through system(). But in addition to the web contents, we also need libcurl to obtain the redirected URL so we can recover the absolute URL of a link with relative path.